src/Calculator.Application/ViewModels/MainWindowViewModel.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System.Collections.ObjectModel;
+using System.Globalization;
+using Calculator.Core.MathService;
+using Calculator.HistoryService;
+using Calculator.HistoryService.Models;
+using Calculator.HistoryService.Models.Enums;
+using CommunityToolkit.Mvvm.ComponentModel;
+using LiveChartsCore;
+using LiveChartsCore.Defaults;
+using LiveChartsCore.SkiaSharpView;
+
+namespace Calculator.Application.ViewModels;
+
+public partial class MainWindowViewModel : ViewModelBase
+{
+    [ObservableProperty]
+    private IHistoryService _historyService;
+
+    private readonly IMathService _mathService;
+
+    [ObservableProperty]
+    private NumberFormatInfo _numberStyle = NumberFormatInfo.InvariantInfo;
+
+    [ObservableProperty]
+    private int _maxExpressionLength = 255;
+
+    [ObservableProperty]
+    private string _expression = string.Empty;
+
+    [ObservableProperty]
+    private bool _isExpressionCorrect = false;
+
+    [ObservableProperty]
+    private double? _variable = null;
+
+    [ObservableProperty]
+    private bool _isExpressionWithVariable = false;
+
+    [ObservableProperty]
+    private bool _isGraphSelected = false;
+
+    [ObservableProperty]
+    private ObservableCollection<ISeries> _series = new()
+    {
+        new LineSeries<ObservablePoint>
+        {
+            GeometryFill = null,
+            GeometryStroke = null,
+            Fill = null
+        }
+    };
+
+    [ObservableProperty]
+    private ObservableCollection<Axis> _xAxes = new(new[] { new Axis() });
+
+    [ObservableProperty]
+    private ObservableCollection<Axis> _yAxes = new(new[] { new Axis() });
+
+    [ObservableProperty]
+    private double? _xMin = -5;
+
+    [ObservableProperty]
+    private double? _xMax = 5;
+
+    [ObservableProperty]
+    private double? _yMin = -1;
+
+    [ObservableProperty]
+    private double? _yMax = 1;
+
+    [ObservableProperty]
+    private HistoryEntry? _selectedHistoryEntry;
+
+    [ObservableProperty]
+    private int _currentTabIndex = 1;
+
+    private bool _saveToHistory = true;
+
+    public MainWindowViewModel(
+        IMathService mathService,
+        IHistoryService historyService
+    )
+    {
+        _mathService = mathService;
+        _historyService = historyService;
+    }
+
+    partial void OnExpressionChanged(string value)
+    {
+        Expression = value.ToLower();
+
+        _mathService.SetExpression(Expression);
+
+        IsExpressionCorrect = _mathService.CheckExpressionValid();
+        IsExpressionWithVariable = IsExpressionCorrect && value.Contains('x');
+    }
+
+    partial void OnSelectedHistoryEntryChanged(HistoryEntry? value)
+    {
+        if (value == null)
+        {
+            return;
+        }
+
+        SelectedHistoryEntry = null;
+        Expression = value.Expression;
+        CurrentTabIndex = value.Type == HistoryEntryType.Math ? 1 : 2;
+
+        if (value.Type == HistoryEntryType.Math)
```

Comment:
value.Type is HistoryEntryType.Math && value.Variable is {} variable...
Varibale = variable